packages/fiori/src/ShellBar.ts

```diff
@@ -808,47 +870,50 @@ class ShellBar extends UI5Element {
 		this._defaultItemPressPrevented = false;
 	}
 
-	_handleCustomActionPress(event) {
-		const refItemId = event.target.getAttribute("data-ui5-external-action-item-id");
+	_handleCustomActionPress(e: MouseEvent) {
+		const target = e.target as HTMLElement;
+		const refItemId = target.getAttribute("data-ui5-external-action-item-id");
 
 		if (refItemId) {
 			const shellbarItem = this.items.find(item => {
 				return item._id === refItemId;
 			});
 
-			const prevented = !shellbarItem.fireEvent("click", { targetRef: event.target }, true);
+			const prevented = shellbarItem!.fireEvent("click", { targetRef: e.target }, true);
```

Comment:
we have a fireEvent with a detail => we need to have a type describing this

You can use this example
the abstract NotificationAction has the event detail type declared and exported 
(same can be done in the ShellBarItem)
https://github.com/SAP/ui5-webcomponents/blob/main/packages/fiori/src/NotificationAction.ts#L101

the event is fired from the parent component (as in the ShellBar)
https://github.com/SAP/ui5-webcomponents/blob/main/packages/fiori/src/NotificationListItemBase.ts#L229